src/java.base/share/classes/sun/security/ssl/CertificateMessage.java

```diff
@@ -386,7 +386,7 @@ private void onCertificate(ServerHandshakeContext shc,
                         ClientAuthType.CLIENT_AUTH_REQUESTED) {
                     // unexpected or require client authentication
                     throw shc.conContext.fatal(Alert.BAD_CERTIFICATE,
-                        "Empty server certificate chain");
+                        "Empty client certificate chain");
```

Comment:
Hm, in tls1.3 it should be certificate_required and in 1.2 handshake_failure for required auth.

rfc8446 6.2 “certificate_required:  Sent by servers when a client certificate is
      desired but none was provided by the client.”
rfc5246 7.4.6 “ If the client does not send any certificates, the
      server MAY at its discretion either continue the handshake without
      client authentication, or respond with a fatal handshake_failure
      alert.”

Reply:
Thanks for raising this point.
I just filed a JBS issue: https://bugs.openjdk.org/browse/JDK-8325079